src/SFA.DAS.AssessorService.Data/Staff/StaffReportRepository.cs

```diff
@@ -42,10 +41,23 @@ public Task<IEnumerable<StaffReport>> GetReportList()
             return (await _connection.QueryAsync(report.StoredProcedure, commandType: CommandType.StoredProcedure)).OfType<IDictionary<string, object>>().ToList();
         }
 
-
         public async Task<IEnumerable<IDictionary<string, object>>> GetDataFromStoredProcedure(string storedProcedure)
         {
-            return (await _connection.QueryAsync(storedProcedure, commandType: CommandType.StoredProcedure)).OfType<IDictionary<string, object>>().ToList();
+            try
+            {
+                var reportId = _assessorDbContext.StaffReports.FirstOrDefault(rep => rep.StoredProcedure == storedProcedure.Trim());
+
+                if (reportId is null)
+                {
+                    return null;
+                }
+
+                return await GetReport(reportId.Id);
+            }
+            catch
+            {
+                return null;
+            }
```

Comment:
A rule of thumb is to avoid catching exceptions if you don't have a good way of handling them. Here, not only is the catch block not handling the error, it's suppressing it and preventing its caller from catching SqlExceptions (on line 96 of StaffReportsController).

I think we should allow it to visibly blow up without catching the exception.